src/jobs/quality_gate_job.yml

```diff
@@ -35,7 +38,24 @@ steps:
         - restore_cache:
             name: restore the dependencies cache
             key: <<parameters.cache_name>>
-  - steps: << parameters.quality_checks >>
+  - when:
+      condition: <<parameters.working_directory>>
+      steps:
+        - run:
+            name: Installing dependencies
+            command: make install
+            working_directory: <<parameters.working_directory>>
+        - run:
+            name: Test that the app can successfully build
+            command: make build
+            working_directory: <<parameters.working_directory>>
+        - run:
+            name: Running tests
+            command: make test
+            working_directory: <<parameters.working_directory>>
+  - unless:
+      condition: <<parameters.working_directory>>
+      steps:  << parameters.quality_checks >>
```

Comment:
Comparing this job and command `quality_gate` I think we should reuse it and make this job simpler.